src/controllers/postController.js

```diff
@@ -4,25 +4,104 @@ import {
   createMetaData,
   getLastPost,
 } from "../repositories/postRepository.js";
+import { getExistingHashtags, insertHashtags, insertHashtagsLinksMiddleTable } from "../repositories/hashtagRepository.js";
 
 export async function postLink(req, res) {
   const { link, postText } = req.body;
   const { user } = res.locals;
-  const {regex} = res.locals;
-
-
+  const { regex } = res.locals;
   try {
     await createPost(link, postText, user.id);
     const { rows: lastPost } = await getLastPost(user.id);
     await createMetaData(lastPost);
-
-    res.sendStatus(201);
+    if(regex.length > 0) postHashtags(lastPost[0].id, res);
```

Comment:
didnt destructure lastpost's array into single object not to mess with it's use by createMetaData function